sql/src/main/java/io/crate/analyze/validator/GroupBySymbolValidator.java

```diff
@@ -101,18 +90,7 @@ public Void visitMatchPredicate(MatchPredicate matchPredicate, Context context)
 
         @Override
         protected Void visitSymbol(Symbol symbol, Context context) {
-            // If is literal no further datatype validation is required
-            if (Symbols.allLiterals(symbol)) {
-                return null;
-            }
-
-            if (context.parentScalar == null) {
-                validateDataType(symbol, context.invalidTypeErrorMessage);
-            } else {
-                validateDataType(context.parentScalar, context.invalidTypeErrorMessage);
```

Comment:
Afaik this was the only place where the `Context` was used so we can remove it safely.